packages/hub/writers.js

```diff
@@ -80,7 +81,7 @@ class Writers {
       if (pending) { await pending.abort();  }
     }
 
-    return schema.applyReadAuthorization(pristine, { session });
+    return context.applyReadAuthorization(pristine, { session });
```

Comment:
@balinterdi same question around applying document context to an arbitrary document vs the pristine document already contained in the document context. I think we are abusing document context by allowing it apply read authorization against documents that are not part of its context.